src/num_bigint.rs

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) 2017-present PyO3 Project and Contributors
+//
+// based on Daniel Grunwald's https://github.com/dgrunwald/rust-cpython
+
+#![cfg(all(feature = "num-bigint", not(any(Py_LIMITED_API, PyPy))))]
+#![cfg_attr(
+    docsrs,
+    doc(cfg(all(feature = "num-bigint", not(any(Py_LIMITED_API, PyPy)))))
+)]
+//!  Conversions to and from [num-bigint](https://docs.rs/num-bigint)’s [`BigInt`] and [`BigUint`] types.
+//!
+//! This is useful for converting Python integers when they may not fit in Rust's built-in integer types.
+//!
+//! # Setup
+//!
+//! To use this feature, add this to your **`Cargo.toml`**:
+//!
+//! ```toml
+//! [dependencies]
+//! num-bigint = "0.4"
+//! pyo3 = { version = "0.14", features = ["num-bigint"] }
+//! ```
```

Comment:
Agreed this is useful to have but I worry we'll forget to update the versions...

Reply:
I don't think there is a way to auto update it :/ This is definitely going to be an annoying maintenance burden going forward.

I've seen other crates document this like:

```toml
[dependencies]
# replace * with the current pyo3 version
pyo3 = { version = "*", features = ["num-bigint"] }
```

Reply:
👍 That seems like a really good idea. Could also say "check pyo3's `Cargo.toml`" for supported `num-bigint` version?

... or we could do something fun with an `mdbook` preprocessor to parse `Cargo.toml` and put the supported `num-bigint` version in the guide in a table 🙈

Reply:
> or we could do something fun with an mdbook preprocessor to parse Cargo.toml and put the supported num-bigint version in the guide in a table 🙈

Is that even possible?

Reply:
I'm not aware of similar tricks to do dynamic docs, however if you want to go for something dynamic in the guide you might find the existing preprocessor I wrote useful: https://github.com/PyO3/pyo3/blob/main/guide/pyo3_version.py

Reply:
This works and will update to the current pyo3 version automagically 🙌
```
//! ```toml
//! [dependencies]
//! num-complex = "*"
#![doc = concat!("pyo3 = { version = \"", env!("CARGO_PKG_VERSION"),  "\", features = [\"num-complex\"] }")]
//! ```
```

Reply:
..but not on stable 😭

It's a syntax related thing so `cfg_attr(docsrs, ..)` won't work..a proc macro maybe? 🤔

Reply:
Since we have other locations where the current version must be substituted, I think the best way is a small (Python) script that replaces everything at once. This can then be run simply as part of the release process.

Reply:
I tried a couple of things but didn't find anything that didn't feel like a gross hack. So I've just taken the `*` approach as above.

Reply:
What's the "gross hack" with my suggestion?

Reply:
It's not, really. I figured out a proc macro that does it by just search+replacing text in a module, but it doesn't work in lib.rs and messes up the [src] links (which is pretty gross).

I don't like the idea of adding things to the release process for something that's at best "nice to have". It's not that big of a deal on its own but these things do add up.

Reply:
TBH I would be happy to have a python script to automate releasing (at the moment it's done manually by me), but i think that's best in a separate PR.